warmup/subtree.py

```diff
@@ -0,0 +1,73 @@
+from anytree import Node, RenderTree 
+from itertools import combinations, product
+'''
```

Comment:
Minor stylistic comment: docstrings in Pyton are just below `def function_name(...) -> ...:`, rather than above it.